fe/fe-core/src/main/java/org/apache/doris/statistics/SQLStatisticsTask.java

```diff
@@ -17,47 +17,119 @@
+        List<StatsType> statsTypes = statsDesc.getStatsTypes();
+        StatsType type = statsTypes.get(0);
+
+        StatsGranularity statsGranularity = statsDesc.getStatsGranularity();
+        Granularity granularity = statsGranularity.getGranularity();
+        boolean nonPartitioned = granularity != Granularity.PARTITION;
+
+        switch (type) {
+            case ROW_COUNT:
+                return nonPartitioned ? InternalSqlTemplate.buildStatsRowCountSql(params)
+                        : InternalSqlTemplate.buildStatsPartitionRowCountSql(params);
+            case NUM_NULLS:
+                return nonPartitioned ? InternalSqlTemplate.buildStatsNumNullsSql(params)
+                        : InternalSqlTemplate.buildStatsPartitionNumNullsSql(params);
+            case MAX_SIZE:
+            case AVG_SIZE:
+                return nonPartitioned ? InternalSqlTemplate.buildStatsMaxAvgSizeSql(params)
+                        : InternalSqlTemplate.buildStatsPartitionMaxAvgSizeSql(params);
+            case NDV:
+            case MAX_VALUE:
+            case MIN_VALUE:
+                return nonPartitioned ? InternalSqlTemplate.buildStatsMinMaxNdvValueSql(params)
+                        : InternalSqlTemplate.buildStatsPartitionMinMaxNdvValueSql(params);
+            case DATA_SIZE:
+            default:
+                throw new DdlException("Unsupported statistics type: " + type);
+        }
     }
 
-    protected List<String> executeQuery(SelectStmt query) {
-        // TODO (ML)
-        return null;
+    protected TaskResult executeQuery(StatisticsDesc statsDesc) throws Exception {
+        StatsGranularity granularity = statsDesc.getStatsGranularity();
+        List<StatsType> statsTypes = statsDesc.getStatsTypes();
+        StatsCategory category = statsDesc.getStatsCategory();
+
+        String dbName = Env.getCurrentInternalCatalog()
+                .getDbOrDdlException(category.getDbId()).getFullName();
+        InternalQuery query = new InternalQuery(dbName, statement);
+        InternalQueryResult queryResult = query.query();
+        List<ResultRow> resultRows = queryResult.getResultRows();
+
+        if (resultRows != null && resultRows.size() == 1) {
+            ResultRow resultRow = resultRows.get(0);
+            List<String> columns = resultRow.getColumns();
+            TaskResult result = createNewTaskResult(category, granularity);
+
+            if (columns.size() == statsTypes.size()) {
+                for (int i = 0; i < columns.size(); i++) {
+                    StatsType statsType = StatsType.fromString(columns.get(i));
+                    result.getStatsTypeToValue().put(statsType, resultRow.getString(i));
+                }
+                return result;
+            }
+        }
+
+        // Statistics statements are executed singly and return only one row data
+        throw new DdlException("Statistics query result is incorrect, " + queryResult);
```

Comment:
do we need to add more info about with stat we collect now?

Reply:
sure